doc/bu-xthin-protocol.md

```diff
@@ -0,0 +1,382 @@
+of missing a transaction. This bears out in practice and a simple linear decay algorithm was developed which alters both the number of 
+elements and the false positive rate. However, not knowing how far out of sync our pools are in practice means we can not calculate the 
+with certainty the probability of a false positive and a memory pool miss which will result in a re-requested transaction, so we need to 
+be careful in not cutting too fine a line. Using this approach significantly reduces the needed size of the bloom filter by 50%.
+
+- Bloom Filter Targeting (Beginning in version 12.1): Bloom Filter targeting which is effective in reducing the size of the bloom filters 
+during times when the memory pool is over run with transactions.  This targeting process is memory pool size independent and yields reliably
+smaller bloom filters. It works essentially by narrowing down the transactions to include in the bloom filter to just those that are most 
+likely to be mined in the next block.
+
+- A 64 bit transaction hash is used instead of the full 256 bit hash to further reduce thinblock size while still preventing hash collisions 
+in the memory pool.  This is done using the GET_XTHIN p2p netmessage.  If the block can not be constructed with the transactions in the mempool
+then a GET_XBLOCKTX netmessage will be requested and will contain a list of the missing transaction hashes to be retreieved.  The returning
+XBLOCKTX message will contain a list of the transactions requested.
+
+- In the event of a 64 bit hash collision the receiving peer will re-request a normal THINBLOCK with the full 256 bit transaction hashes.
+This is done using a GETDATA with an inventory message containing the MSG_THINBLOCK message type.
+
+- In the event that the re-requested THINBLOCK containing the full transaction hashes is not received or is missing transactions then a
+normal block is requested.
+
+
+3. Detailed Protocol Specification
+----------------------------------
+----------------------------------
+
+
+XTHIN Service Bit
+-----------------
+
+defined in protocol.h.  Add the following service:
+
+    /** nServices flags */
+    enum {
+
+        NODE_XTHIN = (1 << 4),
+
+    ... }
+
+    NOTE:  don't forget to update and add the XTHIN service for the bitoin-qt GUI.  Defined in /qt/guiutil.cpp
+    under:
+
+    QString formatServicesStr(quint64 mask)
+
+    add the following:
+
+            case NODE_XTHIN:
+                strList.append("XTHIN");
+                break;
+
+
+
+New p2p  message definitions
+----------------------------
+
+add the following entries defined in protocol.cpp:
+
+namespace NetMsgType {
+
+    ... 
+
+    const char *THINBLOCK="thinblock";
+    const char *XTHINBLOCK="xthinblock";
+    const char *XBLOCKTX="xblocktx";
+    const char *GET_XBLOCKTX="get_xblocktx";
+    const char *GET_XTHIN="get_xthin";
+
+};
```

Comment:
I suggest you don't show C++ code here. Just name them by their protocol string. "thinblock", "xthinblock" etc.

An implementor may not be using C++ nor want to use constants.